tests/integration/testutil/testutil.go

```diff
@@ -0,0 +1,210 @@
+
+	apiv1beta1 "github.com/kubernetes-sigs/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	svcatclient "github.com/kubernetes-sigs/service-catalog/pkg/client/clientset_generated/clientset"
+	"github.com/kubernetes-sigs/service-catalog/pkg/svcat"
+	servicecatalog "github.com/kubernetes-sigs/service-catalog/pkg/svcat/service-catalog"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	interval = time.Second
+	timeout  = time.Minute * 3
+)
+
+// KubeClient creates a new Kubernetes client using the default kubeconfig.
+func KubeClient() (kubernetes.Interface, error) {
+	config, err := kubeConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize kubernetes client: %v", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize kubernetes client: %v", err)
+	}
+
+	return clientset, nil
+}
+
+// SvcatClient creates a new svcat client using the default kubeconfig.
+func SvcatClient() (*svcatclient.Clientset, error) {
+	config, err := kubeConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize svcat client: %v", err)
+	}
+
+	clientset, err := svcatclient.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize svcat client: %v", err)
+	}
+
+	return clientset, nil
+}
+
+func kubeConfig() (*rest.Config, error) {
+	location := os.Getenv("KUBECONFIG")
+	if location == "" {
+		location = filepath.Join(os.Getenv("HOME"), ".kube", "config")
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", location)
+	if err != nil {
+		config, err = rest.InClusterConfig()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get kubernetes configuration: %v", err)
+		}
+	}
+
+	return config, nil
+}
+
+// Svcat wraps the svcat functionality for easier use with the integration tests.
+type Svcat struct {
+	kubeClient kubernetes.Interface
+	client     *svcatclient.Clientset
+	app        *svcat.App
+}
+
+// NewSvcat constructs a new Svcat.
+func NewSvcat(kubeClient kubernetes.Interface, namespace string) (*Svcat, error) {
+	client, err := SvcatClient()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a new Svcat: %v", err)
+	}
+	app, err := svcat.NewApp(kubeClient, client, namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a new Svcat: %v", err)
+	}
+	sc := &Svcat{
+		kubeClient: kubeClient,
+		client:     client,
+		app:        app,
+	}
+	return sc, nil
+}
+
+// WaitForBroker waits for the broker to be ready.
+func (sc *Svcat) WaitForBroker(
+	name string,
+	namespace string,
+) (servicecatalog.Broker, error) {
+	opts := &servicecatalog.ScopeOptions{
+		Scope:     servicecatalog.AllScope,
+		Namespace: namespace,
+	}
+	broker, err := sc.app.WaitForBroker(name, opts, interval, &timeout)
+	if err != nil {
+		return nil, fmt.Errorf("failed to wait for broker: %v", err)
+	}
+
+	return broker, nil
+}
+
+// Provision provisions an instance.
```

Comment:
Please document if this (and deprovision, bind, unbind) does so synchronously or not.

Reply:
Fixed with https://github.com/kubernetes-sigs/minibroker/pull/59/commits/1f5f0cbac0386ebb32f0bfc0ab12c58a833eb39b.